File: SIAN julia/cholera.jl
# Lee, E. C., Kelly, M. R., Ochocki, B. M., Akinwumi, S. M., Hamre, K. E., Tien, J. H., Eisenberg, M. C.,
# Model distinguishability and inference robustness in mechanisms of cholera transmission and loss of immunity
# https://doi.org/10.1016/j.jtbi.2017.01.032
# Eq. (3)
using Logging, SIAN


ode = @ODEmodel(
    S'(t) = mu - bi * S(t) * I(t) - bw * S(t) * W(t) - mu * S(t) + a * R(t),
    I'(t) = bw * S(t) * W(t) + bi * S(t) * I(t) - (gam + mu) * I(t),
    W'(t) = xi * (I(t) - W(t)),
    R'(t) = gam * I(t) - (mu + a) * R(t),
    y1(t) = k * I(t),
    y2(t) = S(t)+I(t)+R(t)
)

@time println(identifiability_ode(ode, get_parameters(ode); p = 0.99, p_mod = 0, nthrds = 1))

println(res)

File: StrikePy/Cholera.py
# Lee, E. C., Kelly, M. R., Ochocki, B. M., Akinwumi, S. M., Hamre, K. E., Tien, J. H., Eisenberg, M. C.,
# Model distinguishability and inference robustness in mechanisms of cholera transmission and loss of immunity
# https://doi.org/10.1016/j.jtbi.2017.01.032
# Eq. (3)

import sympy as sym

S, I, W, R, bi, gam, mu, bw, k, xi, a = sym.symbols('S I W R bi gam mu bw k xi a')

#3 states:
x = [[S], [I], [W], [R]]

# 5 parameters:
p = [[bi], [gam], [mu], [bw], [k], [xi], [a]]

# 2 outputs:
h = [[k*I], [S+R+I]]

# 1 unknown input (time-varying parameter):
u = []
w = []

# dynamic equations:
f = [[-bi*S*I - S*mu - S*bw*W + R*a + mu],
    [bi*S*I - gam*I + S*bw*W - mu*I],
    [xi*I - xi*W],
	[gam*I - R*mu - R*a]]

variables_locales = locals().copy()
